app/api/db/models/example_models.py

```diff
@@ -0,0 +1,42 @@
+from datetime import date
```

Comment:
question for discussion, should we aim to have one model per file like `models/person.py` which exports `Person`, `models/user.py` which exports `User`, or a single module like `models/example_models.py` which has `User` and `Person` in the file? i'm biased towards having each file focus on one thing, but am interested in what others think.

question for my understanding: if we put multiple models in one file, then in practice what would you call that file i.e. what would `example_models.py` be called? `models/models.py` seems redundant.

Reply:
I strongly prefer one model per file.

Reply:
I actually prefer a 3rd option, logically grouped files. 1-per-file seems like a lot of files once a project grows large enough. A quick check of the Mass code shows we have ~200 tables. If I needed to look up something with say payments, I would have needed to open a dozen files to see all the relevant tables.

Lets say you have a `person` file, and a few related tables that would only ever be attached to a person like lookup tables for race/gender/etc. Definitely would want it more split up than the Mass code (the employee model file in particular).

Reply:
Refining my language a bit to use nomenclature from domain driven design, if we define entities and value objects separately, where entities are "Objects that have a distinct identity that runs through time and different representations. You also hear these called "reference objects" and value objects are "Objects that matter only as the combination of their attributes", then things like `person`, `user`, `application` are entities and things like enums/lookup table values and addresses are value objects, one suggestion is that we have one "entity" per file. Related classes for that entity such as any enums could be in the same value.

- https://martinfowler.com/bliki/EvansClassification.html
- https://enterprisecraftsmanship.com/posts/entity-vs-value-object-the-ultimate-list-of-differences/

Reply:
@lorenyu 
> Related classes for that entity such as any enums could be in the same value.

Do you mean in the same file?

Reply:
@chouinar 
> I actually prefer a 3rd option, logically grouped files. 1-per-file seems like a lot of files once a project grows large enough. A quick check of the Mass code shows we have ~200 tables. If I needed to look up something with say payments, I would have needed to open a dozen files to see all the relevant tables.

Yah that's fair and I agree with that too.

Reply:
Might be better for me to think of it like "the big things" or the top-level objects. (And sidenote: it might just be the example used, but I find a lot of flaws with the ideas argued in that second link).

But I think this is getting a bit sidetracked. If I renamed the file to `example_person.py` would that make more sense? The person is the main entity, and pets are a subentity that goes with them.

Reply:
if pet is a top level resource (e.g. we're going to have API routes for them) then i'd have a separate file for it, but if it's only ever relevant in the context of a person, then keeping it in the person file makes sense.

to my point about realistic examples, i don't think the concept of `pet` is close enough to our projects for it to provide a useful reference for when an engineer is determining whether or not to include something like `address` in the same file or not.

Reply:
Happy to adjust the example (maybe something like `ExampleRole`?), but I do think we'd need to add some pretty thorough documentation explaining the organizational philosophy of the models, otherwise a developer will just see "oh hey, you can put the models in multiple files" and they'll just organize arbitrarily.

I'm still not sure if we're in agreement or not on the right way to structure this as reading through the documents you linked above isn't quite clicking with me on the distinction you're trying to make.

Reply:
yeah, I think `User` and `Role` would be great examples, with `Role` and `User` both defined in the `models/user.py` module.

good point about needing documentation regardless

Reply:
I might punt switching to user/role to the ticket where I add more endpoints as we'd need more than one endpoint to have that API make sense (at least a `create role` and `create user` one) and I'd like to keep this contained to purely "making it generic".

Unless you've got any other ideas that might work with the shape I've already created for this endpoint. Was aiming for a "core" record with a list of associated records (in this case, a person has multiple pets) that logically could be done in a single POST.

Reply:
sounds good to me